### DISCUSSION TITLE
Table - 'empty' message visible while loading data

### DISCUSSION THREAD

**Discussion by Malvineous** [sheinsight/shineout#1301] [Q&A]
Hello again!

I have a `<Table/>` which takes a long time to load the data.  I am passing the `loading` prop to display a `Spin` component while it is loading, however the default empty message `Data not found` remains visible under the `Spin`:

![image](https://user-images.githubusercontent.com/171656/137594182-c14bd592-1fc6-4881-b071-236b08db36ed.png)
```
<Table
  loading={true}
  empty={loading ? '' : 'No data'}
/>
```
Do you think the `empty` message should automatically be hidden when `loading` is `true`?

**Comment by EastblueOkay** [comment]
Theoretically speaking, there will be two situations where there is data and there is no data when loading. 
I think it is necessary to give a reminder when there is no data, even if he is still loading.
Maybe our loading view is not obvious enough? Caused to overlap with the contents of the 'Data not found'.

**Reply by Malvineous** [reply]
For the situation where it is loading and there is data, it makes sense to show the existing data and the load animation, that makes sense.

For the situation where it is loading and there is no data, is it ever valuable to show "Data not found" at the same time as the loading animation?  If you removed "Data not found" when the `Spin` was visible, would this lead to confusion?  I think it would not, but I would be interested to hear your opinion.

Once the loading has finished, showing a blank is confusing (not knowing whether the load has finished or not), so I agree having "Data not found" appear once the `Spin` disappears is very useful.

However during the load, when a `Spin` is visible, this gives feedback to the user that something is still in progress.  I think in this case seeing "Data not found" increases confusion, because there are two conflicting UI elements.  The `Spin` tells the user the operation is in progress, but "Data not found" tells the user the operation has completed.  So because of this conflicting information, the user is uncertain whether the operation is in progress or has finished.

If you think it is important to have "Data not found" visible in some situations and leave the decision on whether to hide or show it up to your users that is perfectly fine, however there does not appear to be a neat way to hide the "Data not found" message while the `Spin` is visible.  If I pass `empty={null}` or `empty=""` the message is still visible.  Is there an elegant way to hide this message while the `Spin` is visible?

**Reply by EastblueOkay** [reply]
Empty is set to null or '' still shows that it is indeed a problem. If you don’t mind, you can temporarily set it as an invisible element, For example, ‘’, E.g ' ', <span />.
I think displaying both at the same time may indeed confuse users, Let's modify the default behavior in the next version (Data not found will not be displayed in the loading state) 🤝.

**Reply by Malvineous** [reply]
That sounds perfect, many thanks!  I will use the empty element until the next version!
